sourcemash/frontend/static/js/views/browse_view.js

```diff
@@ -0,0 +1,66 @@
+Sourcemash.Views.BrowseView = Backbone.View.extend({
+  template: JST['browse'],
+
+  initialize: function (options) {
+    var ExtendedTypeahead = Backbone.Typeahead.extend({
+      template: JST['new-feed-form'],
+    });
+
+    this.typeahead = new ExtendedTypeahead({collection: this.collection, key: 'title'});
+    this.listenTo(this.collection, 'sync', this.render);
+  },
+
+  events: {
+    'submit #add_feed_form': 'createFeed',
+  },
+
+  createFeed: function(e){
+  	e.preventDefault()
+    this.collection.create(this.newAttributes(), {success: this.updateCollection});
+  },
+
+  updateCollection: function(newFeed) {
+  	this.$('#url').val('');
+  	newFeed.collection.fetch();
+    toast('Feed added!', 3000)
+
+    mixpanel.track("Subscribed", { "Feed Title": newFeed.get('title'),
+                                    "Source": 'search' })
+  },
+
+  render: function() {
+    // Render parent view
+    this.$el.html(this.template({ models: this.collection.models }));
+    $('#typeahead').html(this.typeahead.render().el);
+
+    // Render item cards
+    var feedCards = [];
+    this.collection.models.forEach(function(feed) {
```

Comment:
Do we need to do this full feedCards approach? Toggle subscribe on one feed won't change any of the other cards ever.